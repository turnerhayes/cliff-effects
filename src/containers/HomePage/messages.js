export default {
  appName: { id: 'homePage.appName' },

  prototypeNote: { id: 'homePage.prototypeNote' },

  cautionaryNote: { id: 'homePage.cautionaryNote' },

  toFirstInputPage: { id: 'homePage.toFirstInputPage' },
  
  toAboutPage: { id: 'homePage.toAboutPage' },
};
