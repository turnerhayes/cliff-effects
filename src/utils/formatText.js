import React from 'react';
import marksy from 'marksy';

const compile = marksy({ createElement: React.createElement });

const formatText = ({ text }) => {
  return compile(text).tree;
};

export default formatText;
