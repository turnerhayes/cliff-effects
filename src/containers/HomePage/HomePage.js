import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

class HomePage extends Component {
  render() {
    return (
      <div id='HomePage'>
        <div id='HomeContent'>
          <div className="main-wrapper">
            <div className="headings">
              <h1>
                <FormattedMessage
                  { ...messages.appName } />
              </h1>
              <h2>
                <FormattedMessage
                  { ...messages.prototypeNote } />
              </h2>
            </div>

            <div className='center-contents'>
              <Link
                className='home-button hb-left'
                to='/visit/54321/1'>
                <FormattedMessage
                  { ...messages.toFirstInputPage } />
              </Link>

              <Link
                className='home-button hb-right'
                to='/about'>
                <FormattedMessage
                  { ...messages.toAboutPage } />
              </Link>
            </div>
            <p className="home-disclaimer">
              <FormattedMessage
                { ...messages.cautionaryNote } />
            </p>
          </div>
        </div>
      </div>
    );
  }
}

export default HomePage;
