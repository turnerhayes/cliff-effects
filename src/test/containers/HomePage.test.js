import React from 'react';
import { mount } from 'enzyme';

import HomePage from '../../containers/HomePage';
import { withRouter } from '../helpers';
import { wrapWithIntlProvider } from '../utils/test-utils';

describe('<HomePage>', () => {
  it('matches snapshot', () => {
    const wrapper = mount(wrapWithIntlProvider(withRouter(<HomePage />)));

    expect(wrapper.children()).toMatchSnapshot();
  });
});
