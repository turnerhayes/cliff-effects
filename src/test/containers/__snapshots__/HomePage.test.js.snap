// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage> matches snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <HomePage>
      <div
        id="HomePage"
      >
        <div
          id="HomeContent"
        >
          <div
            className="main-wrapper"
          >
            <div
              className="headings"
            >
              <h1>
                <FormattedMessage
                  id="homePage.appName"
                  values={Object {}}
                >
                  <span>
                    Cliff Effects Tool
                  </span>
                </FormattedMessage>
              </h1>
              <h2>
                <FormattedMessage
                  id="homePage.prototypeNote"
                  values={Object {}}
                >
                  <span>
                    GUIDANCE PROTOTYPE*
                  </span>
                </FormattedMessage>
              </h2>
            </div>
            <div
              className="center-contents"
            >
              <Link
                className="home-button hb-left"
                replace={false}
                to="/visit/54321/1"
              >
                <a
                  className="home-button hb-left"
                  href="/visit/54321/1"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    id="homePage.toFirstInputPage"
                    values={Object {}}
                  >
                    <span>
                      Get Started
                    </span>
                  </FormattedMessage>
                </a>
              </Link>
              <Link
                className="home-button hb-right"
                replace={false}
                to="/about"
              >
                <a
                  className="home-button hb-right"
                  href="/about"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    id="homePage.toAboutPage"
                    values={Object {}}
                  >
                    <span>
                      Learn More
                    </span>
                  </FormattedMessage>
                </a>
              </Link>
            </div>
            <p
              className="home-disclaimer"
            >
              <FormattedMessage
                id="homePage.cautionaryNote"
                values={Object {}}
              >
                <span>
                  *This is a prototype and should not be used to make financial decisions
                </span>
              </FormattedMessage>
            </p>
          </div>
        </div>
      </div>
    </HomePage>
  </Router>
</MemoryRouter>
`;
